### PR TITLE
Resource switcher

### PR DESCRIPTION
**Pull request by andela-oakinniranye**
This rake task enables the switching of a resource e.g from a comment to an answer or vice versa.

### CODE REVIEW

File: app/services/switch_resource_type.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        resource_attributes= from.as_json.except(*not_needed_attributes).merge(merge_attributes(from))
+        to_resource.create(resource_attributes)
+      end
+    end
+
+    def comments_except
+      ["id", "comment_on_id", "comment_on_type"]
+    end
+
+    def self.answers_except
```
Comment (andela-echigbo) [inline app/services/switch_resource_type.rb]: 👀  This is still within the `class << self` block

File: app/services/switch_resource_type.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      elsif from.instance_of? Answer
+        {comment_on: question}
+      end
+    end
+
+    def permitted_resources
+      [ Comment, Answer ]
+    end
+
+    def method_missing(method_name, *args, &block)
```
Comment (andela-ooranagwa) [inline app/services/switch_resource_type.rb]: `&block`

File: app/services/switch_resource_type.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
+class SwitchResourceType
+  class << self
+    def switch_resource(from_resource, to_resource, from_resource_id)
+      if from= from_resource.find(from_resource_id).destroy
```
Comment (andela-ooranagwa) [inline app/services/switch_resource_type.rb]: Can we have the assignment _cum_ argument to *if* evaluated on a separate line then you might want to consider when `resource` is not found :thought_balloon:

File: app/services/switch_resource_type.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        from_resource, to_resource = [resource_match[:from_resource], resource_match[:to_resource]].map do |resource_type|
+          resource_type.singularize.camelize.constantize rescue nil
+        end
+        raise ResourceSwitchViolation unless ([from_resource, to_resource] & permitted_resources).size == permitted_resources.size
+        switch_resource(from_resource, to_resource, args[0])
+      end
+    end
+
+    def respond_to_missing?(method_name, *args)
+      return true unless method_name.match(/\A(?:answer|comment)_to_(?:answer|comment)\z/)
```
Comment (andela-toduah) [inline app/services/switch_resource_type.rb]: Is this supposed to return true `unless` or `if`?

File: app/services/switch_resource_type.rb (hunk 2 above)
Comment (andela-oakinniranye) [inline app/services/switch_resource_type.rb]: ? :thought_balloon:

File: app/services/switch_resource_type.rb (hunk 4 above)
Comment (andela-oakinniranye) [inline app/services/switch_resource_type.rb]: nice catch. Fixed in the next push

**Pull request merged**